src/Microsoft.Diagnostics.EventFlow.Inputs.ActivitySource/ActivitySourceInput.cs

```diff
@@ -0,0 +1,260 @@
+﻿// ------------------------------------------------------------
+//  Copyright (c) Microsoft Corporation.  All rights reserved.
+//  Licensed under the MIT License (MIT). See License.txt in the repo root for license information.
+// ------------------------------------------------------------
+
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using Microsoft.Extensions.Configuration;
+using Newtonsoft.Json;
+using Validation;
+
+using Microsoft.Diagnostics.EventFlow.Configuration;
+
+namespace Microsoft.Diagnostics.EventFlow.Inputs.ActivitySource
+{
+    public class ActivitySourceInput : IObservable<EventData>, IDisposable
+    {
+        private const int SamplingDecisionCacheFlushThreshold = 1024;
+
+        // Using a static dictionary like this is faster than doing Enum.GetName()
+        private static readonly IDictionary<ActivityKind, string> ActivityKindNames =
+            Enum.GetValues(typeof(ActivityKind)).Cast<ActivityKind>().ToDictionary(k => k, k => k.ToString());
+
+        private EventFlowSubject<EventData> subject_;
```

Comment:
It is a modern C++ naming convention for private members that I ended up adopting. Saves a few keystrokes over camel casing because one does not have to prepend names of class members with `this.` to be able to immediately tell what is a class member vs. local variable or function parameter.

I can revert to camel case if you feel strongly about it.

Reply:
Modern C++ not mordern C#? :)
I think C# just doesn't want the feature to quickly tell local variable, function parameter or private member, I think compiler will give suggestion saying "this" can be removed to simplify the code. And I'm in favor of consistency. I'm not so strong about it though.

Reply:
> (...) I think compiler will give suggestion saying "this" can be removed to simplify the code.(...)

Which is exactly what I **don't** want because it reduces the readability of the code IMO 🙄

I will keep this as-is for now.